libraries/AP_Logger/LoggerMessageWriter.h

```diff
@@ -145,6 +145,9 @@ class LoggerMessageWriter_DFLogStart : public LoggerMessageWriter {
 
 private:
 
+    // check for using too much time
+    static bool check_process_limit(uint32_t start_us);
```

Comment:
I think I could see `static` in the base class (`LoggrMessageWriter) to try to stop long loops overall.  But we're passing the time in.  But this could be `const`, too, so I'm thinking this was formulated differently earlier?

```suggestion
    bool check_process_limit(uint32_t start_us) const;
```

Reply:
static means we save passing in the class ptr, its a small optimisation for functions doing pure computation